esmvalcore/_recipe.py

```diff
@@ -361,6 +362,51 @@ def _get_default_settings(variable, config_user, derive=False):
     return settings
 
 
+def _add_fxvar_keys(fx_var_dict, variable):
+    """Add keys specific to fx variable to use get_input_filelist."""
+    fx_variable = dict(variable)
+
+    # set variable names
+    fx_variable['variable_group'] = fx_var_dict['short_name']
+    fx_variable['short_name'] = fx_var_dict['short_name']
+
+    # specificities of project
+    if fx_variable['project'] == 'CMIP5':
+        fx_variable['mip'] = 'fx'
+        fx_variable['ensemble'] = 'r0i0p0'
+    elif fx_variable['project'] == 'CMIP6':
+        fx_variable['grid'] = variable['grid']
+        if 'mip' in fx_var_dict:
+            fx_variable['mip'] = fx_var_dict['mip']
+    # add missing cmor info
+    _add_cmor_info(fx_variable, override=True)
+
+    return fx_variable
+
+
+def _get_correct_fx_file(variable, fx_varname, config_user):
+    """Wrapper to standard file getter to recover the correct fx file."""
+    var = dict(variable)
+    if var['project'] == 'CMIP5':
+        fx_var = _add_fxvar_keys({'short_name': fx_varname, 'mip': 'fx'}, var)
+    elif var['project'] == 'CMIP6':
+        if fx_varname == 'sftlf':
+            fx_var = _add_fxvar_keys({'short_name': fx_varname, 'mip': 'fx'},
+                                     var)
+        elif fx_varname == 'sftof':
+            fx_var = _add_fxvar_keys({'short_name': fx_varname, 'mip': 'Ofx'},
+                                     var)
+        # TODO check this is the correct mip
+        elif fx_varname == 'sftgif':
+            fx_var = _add_fxvar_keys({'short_name': fx_varname, 'mip': 'Efx'},
+                                     var)
```

Comment:
cheers dude! I changed the mip to fx and changed the TODO as well

Reply:
:+1: Is it possible to specify a mip explicitly if I as a user want to do that in the recipe?

Reply:
that could be easily implemented as argument to the preproc masking function but I think keeping things automated is much better in this case since we know that either sftlf or sftof or sftgif are needed; I reckon this should be addressed in another PR if really needed

Reply:
I think we could add a list of mips instead of a single mip, that would work so long as there is only a single file

Reply:
Unfortunately, usually the choice is to have it either in one {,O,E,...}fx table or in both {,O,E,...}mon and dec tables. But let's not worry about it for now and just get on with the dominant fx case.